### DISCUSSION TITLE
Obtaining Path Metadata for Asset Modules

### DISCUSSION THREAD

**Discussion by nsunga** [webpack/webpack#17647] [Q&A]
Hello Webpack people,

Lets say I have `foo.jpg` and it lives in `my-project/images/foo.jpg`

And my Webpack output path is `dist/`

Now lets process this with a Webpack rule like this:
```
rules: [
    {
      test: /\.(jpg)$/i,
      type: 'asset/resource',
      generator: {
        filename: '[name].[hash][ext]',
      },
    },
]
```

In my `dist/` folder, I'll now have the processed image. So something like this `foo.jpg => foo.123456.jpg`

**So my question is this:** Is it possible that I can obtain the absolute path of `foo.jpg` and its Webpack equivalent either during Webpack build, or post-build with a plugin? If yes, which source code files should I be looking at?

Some context: Im looking for ways to server side render these asset modules - Im using babel on the server side. So, Im not bundling the server with Webpack (which I think would solve my issue, but its a longer solution to implement for a business) and the issue is locating the Webpack bundled assets like images.

**Comment by alexander-akait** [comment]
You can use https://github.com/shellscape/webpack-manifest-plugin
